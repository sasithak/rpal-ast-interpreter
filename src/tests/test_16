let 
.function_form
..<ID:merge>
..<ID:x>
..<ID:y>
..where
...->
....not
.....gamma
......<ID:Isstring>
......<ID:x>
....<STR:'x should be a string'>
....->
.....not
......gamma
.......<ID:Isstring>
.......<ID:y>
.....<STR:'y should be a string'>
.....->
......eq
.......<ID:x>
.......<STR:''>
......<ID:y> 
......->
.......eq
........<ID:y>
........<STR:''>
.......<ID:x>
.......gamma
........gamma
.........<ID:merge_r>
.........<ID:x>
........<ID:y>
...rec
....function_form
.....<ID:merge_r>
.....<ID:x>
.....<ID:y>
.....->
......eq
.......<ID:x>
.......<STR:''>
......<ID:y>
......->
.......eq
........<ID:y>
........<STR:''>
.......<ID:x>
.......->
........le
.........<ID:x>
.........<ID:y>
........gamma
.........gamma
..........<ID:Conc>
..........gamma
...........<ID:Stem>
...........<ID:x>
.........gamma
..........gamma
...........<ID:merge_r>
...........gamma
............<ID:Stern>
............<ID:x>
..........<ID:y>
........gamma
.........gamma
..........<ID:Conc>
..........gamma
...........<ID:Stem>
...........<ID:y>
.........gamma 
..........gamma
...........<ID:merge_r>
...........<ID:x>
..........gamma
...........<ID:Stern>
...........<ID:y>
.gamma
..<ID:Print>
..tau
...gamma
....gamma
.....<ID:merge>
.....<STR:'acegh'>
....<STR:'bdfi'>
...gamma
....gamma
.....<ID:merge>
.....<STR:'ilps'>
....<STR:'akltv'>
...gamma
....gamma
.....<ID:merge>
.....<STR:''> 
....<STR:'bcde'>
...gamma
....gamma
.....<ID:merge>
.....<STR:'abcd'>
....<STR:''>
...gamma
....gamma
.....<ID:merge>
.....<INT:1>
....<STR:'bcde'>
...gamma
....gamma
.....<ID:merge>
.....<STR:'abcd'>
....<INT:1>
