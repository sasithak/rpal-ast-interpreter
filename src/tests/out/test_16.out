(abcdefghi, aikllpstv, bcde, abcd, x should be a string, y should be a string)
